fmt/container.h

```diff
@@ -0,0 +1,81 @@
+/*
+ Formatting library for C++ - standard container utilities
+
+ Copyright (c) 2012 - 2016, Victor Zverovich
+ All rights reserved.
+
+ For the license information refer to format.h.
+ */
+
+#ifndef FMT_CONTAINER_H_
+#define FMT_CONTAINER_H_
+
+#include "format.h"
+
+namespace fmt {
+
+namespace internal {
+
+/**
+  \rst
+  A "buffer" that appends data to a standard container (e.g. typically a
+  ``std::vector`` or ``std::basic_string``).
+  \endrst
+ */
+template <typename Container>
+class ContainerBuffer : public Buffer<typename Container::value_type> {
+ private:
+  Container& container_;
+  std::size_t initial_size_;
+
+ protected:
+  virtual void grow(std::size_t size) FMT_OVERRIDE {
+    container_.resize(initial_size_ + size);
+    this->ptr_ = &container_[initial_size_];
+    this->capacity_ = size;
+  }
+
+ public:
+  explicit ContainerBuffer(Container& container)
+  : container_(container), initial_size_(container.size()) {}
```

Comment:
`ptr_` points to the beginning of the buffer (container's data), `size_` is the size of the filled part of the buffer and `capacity_` is the size + free space. So if you initialize `size_` append should Just Work, because the new output operations will write past the initial size, unless I'm missing something myself.